src/halo-hooks/useRewards.ts

```diff
@@ -307,3 +314,73 @@ export const useRewarderUSDPrice = (rewarderAddress: string | undefined) => {
 
   return rewarderTokenUsdPrice[rewarderTokenAddress[0]]
 }
+
+export const useGetBaseApr = (poolAddress: string, tokenPair: string): number => {
+  const { account, library, chainId } = useActiveWeb3React()
+  const [baseAPR, setBaseApr] = useState(0)
+  const poolsWithBaseAPR = Object.keys(DEPOSIT_TXHASH)
+
+  const fetchBaseUsdHlpPrice = useCallback(async () => {
+    if (!library || !poolsWithBaseAPR.includes(poolAddress.toLowerCase())) return
+
+    // get the first deposit transaction of the curve
+    const txReceipt = await library.getTransactionReceipt(DEPOSIT_TXHASH[poolAddress.toLowerCase()])
+
+    // get the liquidity and total supply of the curve
+    const curveContract = await getContract(poolAddress, CURVE_ABI, library, account ?? undefined)
+    const [curveLiquidity, curveTotalSupply] = await Promise.all([
+      curveContract.liquidity(),
+      curveContract.totalSupply()
+    ])
+    const totalCurveLiquidity = parseFloat(formatEther(curveLiquidity.total_))
+    const totalCurveSupply = parseFloat(formatEther(curveTotalSupply))
+
+    // get the date of the deposit transaction and the number of days since the tx
+    const txTimestamp = (await library.getBlock(txReceipt.blockNumber)).timestamp
+    const date = getTxDate(txTimestamp)
+    const dateNow = new Date().getTime()
+    const noOfDaysSinceFirstDeposit = Math.floor((dateNow - txTimestamp * 1000) / (1000 * 60 * 60 * 24))
+
+    // get the base token symbol and decimals
+    const baseTokenSymbol = tokenPair.split('/')[0]
+    const { decimals } = (haloTokenList[chainId as ChainId] as Token[]).filter(
+      token => token.symbol === baseTokenSymbol
+    )[0]
+
+    // get the transfer logs
+    const logs = txReceipt.logs.filter(
+      (value: any) => value.topics[0] === '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'
```

Comment:
@geonellov I assume `0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef` is the "transfer" event hash? If so, can we put this in a constant file and reference it here as a const instead?

```suggestion
      (value: any) => value.topics[0] === TRANSFER_EVENT_HASH
```